src/sentry/monitors/consumers/monitor_consumer.py

```diff
@@ -1041,8 +1042,6 @@ def shutdown(self) -> None:
             self.parallel_executor.shutdown()
 
     def create_parallel_worker(self, commit: Commit) -> ProcessingStrategy[KafkaPayload]:
-        self.parallel_executor = ThreadPoolExecutor(max_workers=self.max_workers)
```

Comment:
This breaks typing, we need to define `self.parallel_executor` in the class as 

```py
parallel_executor: Executor | None
```

and here we need to do

```py
assert self.parallel_executor
```